src/overlays/actors/ovl_En_Osk/z_en_osk.c

```diff
@@ -33,42 +34,526 @@ const ActorInit En_Osk_InitVars = {
+                break;
+
+            case 10:
+                func_80BF5E00(this, D_80BF6FE4, 9, -5.0f);
+                break;
+
+            case 12:
+                func_80BF5E00(this, D_80BF6FE4, 11, -5.0f);
+                break;
+        }
+    }
+
+    if (func_800EE29C(globalCtx, this->unk_258)) {
+        func_800EDF24(&this->actor, globalCtx, func_800EE200(globalCtx, this->unk_258));
+        if (this->unk_256 != globalCtx->csCtx.npcActions[func_800EE200(globalCtx, this->unk_258)]->unk0) {
+            this->unk_256 = globalCtx->csCtx.npcActions[func_800EE200(globalCtx, this->unk_258)]->unk0;
+            func_80BF67A8(this);
+        }
+        func_80BF68E0(this);
+    } else {
+        this->actor.draw = NULL;
+    }
 
-extern UNK_TYPE D_06006808;
+    if (this->unk_256 == 8) {
+        if (this->actor.scale.x > 0.65f * 0.001f) {
+            this->actor.scale.x -= 0.65f * 0.001f;
+            Actor_SetScale(&this->actor, this->actor.scale.x);
+            func_80BF5EBC(this, globalCtx);
+            func_800B9010(&this->actor, NA_SE_EN_COMMON_EXTINCT_LEV - SFX_FLAG);
+        } else {
+            this->actor.draw = NULL;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/EnOsk_Init.s")
+void EnOsk_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsk* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/EnOsk_Destroy.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF5E00.s")
+void func_80BF6C54(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnOsk* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF5E68.s")
+    if (limbIndex == 1) {
+        Matrix_MultiplyVector3fByState(&D_80BF7024, &this->actor.focus.pos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF5EBC.s")
+void EnOsk_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnOsk* this = THIS;
+    Gfx* gfx;
+    Vec3f sp80;
+    s32 pad2[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF5F60.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF5F70.s")
+    func_8012C28C(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF609C.s")
+    gSPSegment(POLY_OPA_DISP++, 0x08, Gfx_PrimColor(globalCtx->state.gfxCtx, 0x80, 255, 255, 255, 255));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF61EC.s")
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          NULL, func_80BF6C54, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Osk/func_80BF6314.s")
+    {
+        s16 sp62;
+        s16 sp60;
+        Vec3f sp54;
+        sp54 = GET_ACTIVE_CAM(globalCtx)->eye;
```

Comment:
Could you add an extra line here?
```suggestion
        Vec3f sp54;

        sp54 = GET_ACTIVE_CAM(globalCtx)->eye;
```

Also, yikes to this whole scope